### PR TITLE
PEP 685: Comparison of extra names for optional distribution dependencies

### PR DESCRIPTION
**Pull request by brettcannon**

### CODE REVIEW

**Comment by brettcannon** [comment]
I will update the Discussions-To link to a proper one on discuss.python.org once this is passing CI.

File: pep-0685.rst [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+`Setuptools does normalization <https://github.com/pypa/setuptools/blob/b2f7b8f92725c63b164d5776f85e67cc560def4e/pkg_resources/__init__.py#L1324-L1330>`__
+via ``re.sub('[^A-Za-z0-9.-]+', '_', name).lower()``.
+The use of an underscore/``_`` differs from PEP 503's use of a
+hyphen/``-``.
+Runs of characters, unlike PEP 503, do **not** get collapsed,
+e.g. ``___`` stays the same.
+
+For pip, its
+"extra normalisaton behaviour is quite convoluted and eratic",
```
Comment (JelleZijlstra) [inline pep-0685.rst:60]:
```suggestion
"extra normalisaton behaviour is quite convoluted and erratic",
```

File: pep-0685.rst [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+and so its use is not considered.
+
+
+Specification
+=============
+
+[Describe the syntax and semantics of any new language feature.]
```
Comment (JelleZijlstra) [inline pep-0685.rst:67]: Remove this line

File: pep-0685.rst [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+``re.sub('[^A-Za-z0-9.-]+', '_', name).lower()``.
+This normalizes any extra name previously allowed by :pep:`508` in a
+consistent fashion with setuptools.
+
+For tools writing `core metadata`_,
+they MUST write out extra names in their normalized form.
+This applies to the ``Provides-Extra`` field and the ``Provides-Dist``
+field both when specifying extras for a distribution as well as the
+``extra`` marker.
+This will also help enforce the curren requirement from the core
```
Comment (JelleZijlstra) [inline pep-0685.rst:80]:
```suggestion
This will also help enforce the current requirement from the core
```

File: pep-0685.rst [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+
+Security Implications
+=====================
+
+It is possible that a distribution has conflicting extra names and a
+tool ends up installing distributions that somehow weaken the security
+of the system.
+This is only hypothetical and if it were to occur it would probably be
+more of a security concern for the distributions involved more than
```
Comment (JelleZijlstra) [inline pep-0685.rst:108]:
```suggestion
more of a security concern for the distributions involved than
```
The whole sentence isn't easy to follow though.

**Pull request merged**